src/controllers/transaction/index.ts

```diff
@@ -0,0 +1,56 @@
+import { Controller, Get, Route, Path, Response, Tags, Security, Query } from 'tsoa'
+import type { Logger } from 'pino'
+import { logger } from '../../lib/logger'
+import Database from '../../lib/db'
+import { UUID } from '../../models/uuid'
+import { BadRequest, NotFound } from '../../lib/error-handler/index'
+import { TransactionResponse, TransactionType } from '../../models/transaction'
+
+@Route('transaction')
+@Tags('transaction')
+@Security('bearerAuth')
+export class TransactionController extends Controller {
+  log: Logger
+  db: Database
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/transaction' })
+    this.db = new Database()
+  }
+
+  /**
+   * Returns the details of all transactions.
+   * @summary List all transactions
+   * @Query apiType lists all transactions by that type
+   */
+  @Response<BadRequest>(400, 'Request was invalid')
+  @Response<NotFound>(404, 'Item not found')
+  @Get('/')
+  public async getAllTransactions(@Query() apiType?: TransactionType): Promise<TransactionResponse[]> {
```

Comment:
the query param says `apiType` but it's using `TransactionType` instead?

Reply:
Yes, this got me for a little while, since I was looking at the tests when just started. If I would have done this earlier would have saved some time:), please confirm that I use the correct type.